constraint/pkg/client/constraint_client.go

```diff
@@ -0,0 +1,76 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/open-policy-agent/frameworks/constraint/pkg/client/errors"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/core/constraints"
+	"github.com/open-policy-agent/frameworks/constraint/pkg/types"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// constraintClient handler per-Constraint operations.
+//
+// Not threadsafe.
+type constraintClient struct {
+	// constraint is a copy of the original Constraint added to Client.
+	constraint *unstructured.Unstructured
+
+	// enforcementAction is what should be done if the Constraint is violated or
+	// fails to run on a review.
+	enforcementAction string
+
+	// matchers are the per-target Matchers for this Constraint.
+	matchers map[string]constraints.Matcher
+}
+
+func (c *constraintClient) getConstraint() *unstructured.Unstructured {
+	return c.constraint.DeepCopy()
+}
+
+func (c *constraintClient) matches(target string, review interface{}) *constraintMatchResult {
+	matcher, found := c.matchers[target]
+	if !found {
+		return nil
+	}
+
+	matches, err := matcher.Match(review)
+
+	switch {
+	case err != nil:
+		// Run the Constraint's enforcementAction since we were unable to determine
```

Comment:
nit: CF doesn't run enforcement actions, enforcement points would. Though we do return the enforcement action.

Reply:
Updated comment.